### DISCUSSION TITLE
Display LILYGO T5 ePaper 4.7 not working

### DISCUSSION THREAD

**Discussion by wlermer** [arendst/Tasmota#13137] [Q&A] [answered]
I'm quite new to tasmota integrations on ESP32 and tried to get a  **LILYGO T5 ePaper 4.7** (http://www.lilygo.cn/prod_view.aspx?TypeId=50031&Id=1384&FId=t3:50031:3) working.
but I cannot get it running and kindly ask you for support... 

I created a new environment setting for my firmware:
```
[env:tasmota32-ttgo47-display]
extends                 = env:tasmota32_base
board                   = esp32_16M
build_flags             = ${env:tasmota32_base.build_flags} -DFIRMWARE_TTGO47_DISPLAY
lib_extra_dirs          = lib/libesp32, lib/lib_basic, lib/lib_display, lib/libesp32_epdiy
```

and added the definitions in **user_config_override.h**
```
#ifdef FIRMWARE_TTGO47_DISPLAY
    // -- CODE_IMAGE_STR is the name shown between brackets on the 
    //    Information page or in INFO MQTT messages
    #undef CODE_IMAGE_STR
    #define CODE_IMAGE_STR "ttgo47-display"

#define USE_BMP
#define USE_I2C
#define USE_SPI
#define USE_DISPLAY
#define USE_LILYGO47
#define SHOW_SPLASH

#endif // #ifdef FIRMWARE_TTGO47_DISPLAY  
```
the build could be deployed successfully but calling `Display ` returns

```
00:34:50.502 CMD: display
00:34:50.518 MQT: stat/tasmota_5D97A8/RESULT = {"Display":{"Model":12,"Type":0,"Width":240,"Height":240,"Mode":0,"Dimmer":7,"Size":1,"Font":1,"Rotate":0,"Invert":0,"Refresh":2,"Cols":[16,8],"Rows":2}}
```
I would expect the display model to be 16 according to the definition in the driver file (and setting `DisplayModel 16`  doesn't work, too)

**xdsp_16_esp32_epaper_47.ino**
```
...
#ifdef ESP32
#ifdef USE_DISPLAY
#ifdef USE_LILYGO47

#define XDSP_16                16
...

```

It even seems to me that this file is not even included in the build... comparing to other displays I'm missing something like this in e.g. **support_features.ino**

```
#if defined(USE_SPI) && defined(USE_DISPLAY) && defined(USE_DISPLAY_ST7789)
    feature7 |= 0x00080000;  // xdsp_12_ST7789.ino
#endif
```

Do I have to configure specific GPIOs in the template for this device?
I'm quite lost now and appreciate any help or hints...

Thank you!

**Comment by Jason2866** [comment]
It seems there is nothing missing so far. There is a hardcoded gpio config here https://github.com/arendst/Tasmota/blob/development/lib/libesp32_epdiy/src/epd_driver/display_ops.h
Maybe you have a new board variant. Try updating the driver https://github.com/vroland/epdiy You have to add a define for your board in  `display_ops.h` -> `#define CONFIG_EPD_BOARD_REVISION_LILYGO_T5_47`

You can try to add to `user_config_override.h`
```
#define CONFIG_EPD_DISPLAY_TYPE_ED047TC1
#define CONFIG_EPD_BOARD_REVISION_LILYGO_T5_47
```

**Comment by Jason2866** [comment]
There is a example for the driver (we use  in Tasmota) https://github.com/vroland/epdiy/tree/master/examples/lilygo-t5-47-epd-platformio

**Comment by Jason2866** [comment]
I ordered the device. In a few weeks i can tell probably more.

**Comment by wlermer** [comment, marked as answer]
Thanks for your suggestions - I found it out now by debugging and serial outputs... the display couldn't be initialized because GPIO_EDP_DATA wasn't set in the template. So finally you have to configure on IO GPIO 23 --> EDP_DATA and it works then.

**Comment by wlermer** [comment]
Sorry for re using this discussion - as the original question was answered. I tried to display a rgb file on the screen but it didn't work. Is this currently not supported?

`displaytext [x10y10P/sun.rgb]`

I created an uncompressed rgb file from an icon using gimp...
Thanks again for your support...
